### PR TITLE
Improves few tests and an assert method

### LINKED ISSUE #1440: Update try/catch test logic in remaining acceptance test suites

### PR DESCRIPTION
**Pull request by konstantinabl**
**Description**:
This PR improves testing in rateLimiter.spec and subscribe.spec, as well as improving the assertPredfeinedJsonRPC assertion

**Related issue(s)**:

Fixes #1440 

### Proposed Changes

- Removes try catch logic in rateLimiter.spec and subscribe.spec files
- Fixes some linting issues

**Checklist**
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/server/tests/acceptance/rateLimiter.spec.ts [hunk 1]
```diff
@@ -54,22 +53,19 @@ describe('@ratelimiter Rate Limiters Acceptance Tests', function () {
...
                 for (let index = 0; index < TIER_2_RATE_LIMIT * 2; index++) {
                     await relay.call(testConstants.ETH_ENDPOINTS.ETH_CHAIN_ID, [null], requestId);
                     // If we don't wait between calls, the relay can't register so many request at one time. So instead of 200 requests for example, it registers only 5.
                     await new Promise(r => setTimeout(r, 1));
                 }
-            } catch (error) {
-                rateLimited = true;
-                Assertions.jsonRpcError(error, predefined.IP_RATE_LIMIT_EXCEEDED(testConstants.ETH_ENDPOINTS.ETH_CHAIN_ID));
             }
+            const error = predefined.IP_RATE_LIMIT_EXCEEDED(testConstants.ETH_ENDPOINTS.ETH_CHAIN_ID)
```
Comment (ebadiere) [inline packages/server/tests/acceptance/rateLimiter.spec.ts:63]: Nit: semi-colon can be added at the end of this line as well as the line above it.

File: packages/server/tests/acceptance/ws/subscribe.spec.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
 // external resources
 import { solidity } from "ethereum-waffle";
-import chai, {expect} from "chai";
+import chai, {Assertion, expect} from "chai";
```
Comment (ebadiere) [inline packages/server/tests/acceptance/ws/subscribe.spec.ts:23]: The 'Assertion' is not used.

File: packages/server/tests/acceptance/ws/subscribe.spec.ts [hunk 3]
```diff
@@ -709,46 +710,21 @@ describe('@web-socket Acceptance Tests', async function() {
...
-            } catch (e: any) {
-                actualError = JSON.parse(e.response);
-            }
-
             const expectedError = predefined.INVALID_PARAMETER(`filters.address`, `${missingContract} is not a valid contract type or does not exists`);
-            expect(actualError.error.code).to.be.eq(expectedError.code);
-            expect(actualError.error.name).to.be.eq(expectedError.name);
-            expect(actualError.error.message).to.contains(expectedError.message);
+
+            await Assertions.assertPredefinedRpcError(expectedError, wsProvider.send, true, wsProvider, ['eth_subscribe', ["logs", {"address": missingContract}]])
```
Comment (ebadiere) [inline packages/server/tests/acceptance/ws/subscribe.spec.ts:715]: Nit: Semi-colon can be added at the end of this statement as well.

File: packages/server/tests/acceptance/ws/subscribe.spec.ts [hunk 4]
```diff
@@ -709,46 +710,21 @@ describe('@web-socket Acceptance Tests', async function() {
...
-            } catch (e: any) {
-                actualError = JSON.parse(e.response);
-            }
-
             const expectedError = predefined.INVALID_PARAMETER(`'address' for EthSubscribeLogsParamsObject`, `Expected 0x prefixed string representing the address (20 bytes) or an array of addresses, value: `);
-            expect(actualError.error.code).to.be.eq(expectedError.code);
-            expect(actualError.error.name).to.be.eq(expectedError.name);
-            expect(actualError.error.message).to.contains(expectedError.message);
+
+            await Assertions.assertPredefinedRpcError(expectedError, wsProvider.send, true, wsProvider, ['eth_subscribe', ["logs", {"address": missingContract}]])
```
Comment (ebadiere) [inline packages/server/tests/acceptance/ws/subscribe.spec.ts:721]: Nit: Semi-colon can be added at the end of this statement as well.

File: packages/server/tests/acceptance/ws/subscribe.spec.ts [hunk 5]
```diff
@@ -709,46 +710,21 @@ describe('@web-socket Acceptance Tests', async function() {
...
-            } catch (e: any) {
-                actualError = JSON.parse(e.response);
-            }
-
             const expectedError = predefined.INVALID_PARAMETER(`'topics' for EthSubscribeLogsParamsObject`, `Expected an array or array of arrays containing 0x prefixed string representing the hash (32 bytes) of a topic, value: 0x000`);
-            expect(actualError.error.code).to.be.eq(expectedError.code);
-            expect(actualError.error.name).to.be.eq(expectedError.name);
-            expect(actualError.error.message).to.contains(expectedError.message);
+
+            await Assertions.assertPredefinedRpcError(expectedError, wsProvider.send, true, wsProvider, ['eth_subscribe', ["logs", {"address": logContractSigner.address, "topics": ["0x000"]}]])
```
Comment (ebadiere) [inline packages/server/tests/acceptance/ws/subscribe.spec.ts:727]: Nit: Semi-colon can be added at the end of this statement as well.

**Review by georgi-l95 — APPROVED** [review]
LG

**Review by Nana-EC — APPROVED** [review]
LG

**Review by AlfredoG87 — APPROVED** [review]
LGTM

**Pull request merged**